services/storage-users/pkg/command/uploads.go

```diff
@@ -311,3 +324,181 @@ func buildInfo(filter storage.UploadSessionFilter) string {
+				spaceID := ctx.String("spaceid")
+				spaceIDs = append(spaceIDs, spaceID)
+			} else {
+				fmt.Println("Scanning all spaces for stale processing nodes...")
+				spaceIDs = globSpaceIDs(cfg)
+			}
+
+			fmt.Printf("Spaces to cleanup: %d\n", len(spaceIDs))
+			for _, spaceID := range spaceIDs {
+				fmt.Printf("  - %s\n", spaceID)
+			}
+
+			for _, spaceID := range spaceIDs {
+				deleteStaleUploads(ctx, cfg, spaceID)
+			}
+
+			fmt.Printf("Took %ds\n", int(time.Since(start).Seconds()))
+
+			return nil
+		},
+	}
+}
+
+// globSpaceIDs returns a list of all space IDs in the storage root
+func globSpaceIDs(cfg *config.Config) []string {
+	nodesDir := "nodes"
+	spaceIDs := []string{}
+	spacesRoot := filepath.Join(cfg.Drivers.OCIS.Root, "spaces")
+
+	// Find all dirs in spacesRoot that have "nodes" inside
+	err := filepath.Walk(spacesRoot, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error accessing path %s: %v\n", path, err)
+			return filepath.SkipDir
+		}
+
+		if info.IsDir() && info.Name() == nodesDir {
+			// For path i.e. ~/.ocis/storage/users/spaces/9d/408cec-8f0a-4d33-8715-89df1217a10c/nodes
+			// spaceID is 9d408cec-8f0a-4d33-8715-89df1217a10c
+			spaceID := strings.ReplaceAll(strings.TrimSuffix(strings.TrimPrefix(path, filepath.Join(spacesRoot, "/")), "/"+nodesDir), "/", "")
+			spaceIDs = append(spaceIDs, spaceID)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error walking spaces root directory %s: %v\n", spacesRoot, err)
+		return nil
+	}
+
+	return spaceIDs
+}
+
+// delete stale processing nodes for a given spaceID
+func deleteStaleUploads(ctx *cli.Context, cfg *config.Config, spaceID string) {
+	fmt.Printf("\nDeleting stale processing nodes for space: %s\n", spaceID)
+
+	// Find .mpk files in space directory
+	spaceRoot := filepath.Join(cfg.Drivers.OCIS.Root, "spaces", lookup.Pathify(spaceID, 1, 2))
+	mpkFiles := []string{}
+	err := filepath.Walk(spaceRoot, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error accessing path %s: %s\n", path, err)
+			return filepath.SkipDir
+		}
+		if !info.IsDir() && strings.HasSuffix(path, ".mpk") {
+			mpkFiles = append(mpkFiles, path)
+		}
+		return nil
+	})
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error walking space directory %s: %s\n", spaceRoot, err)
+		return
+	}
+
+	fmt.Printf("%sFound total %d .mpk files\n", LOG_INDENT_L1, len(mpkFiles))
+
+	staleCount := 0
+	for _, path := range mpkFiles {
+		staleCount += deleteStaleNode(ctx, cfg, path)
+	}
+
+	fmt.Printf("%sFound total %d stale nodes\n", LOG_INDENT_L1, staleCount)
+}
+
+// deleteStaleNode deletes a stale node if it is not referenced by any upload session
+// returns 1 if the node was deleted, 0 otherwise, for counting purposes
+func deleteStaleNode(ctx *cli.Context, cfg *config.Config, path string) int {
+	dryRun := ctx.Bool("dry-run")
+	verbose := ctx.Bool("verbose")
```

Comment:
I wouldn't advise to pass the `cli.Context` so far down the line. This will bind this function to a usage in cli context. Pass a `context.Context` (if you must). Explicitly pass `dry-run` and `verbose` as function parameters. Use the `cli.Context` only in the main `DeleteStaleProcessingNodes` function.